File: Tests/executors/CXX/core/src/TokenGiverExecutor.cpp
#include "TokenGiverExecutor.h"

#include <Utility/WebFrameworkUtility.hpp>

void TokenGiverExecutor::doPost(framework::HttpRequest& request, framework::HttpResponse& response)
{
	using namespace std::chrono_literals;

	framework::JsonBuilder result;
	framework::JsonObject data;

	data["userName"] = request.getJson().get<std::string>("userName");

	result["token"] = framework::utility::token::createJwt(data, 60min);

	response.setBody(result);
}

DEFINE_EXECUTOR(TokenGiverExecutor)

File: Tests/executors/CSharp/CoreExecutors/TokenGiverExecutor.cs
using Framework;
using Framework.Utility;
using System.Text.Json.Nodes;

public class TokenGiverExecutor : StatelessExecutor
{
	public override void DoPost(HttpRequest request, HttpResponse response)
	{
		JsonObject temp = [];

		temp["userName"] = request.GetJson()["userName"].GetString();

		response.SetBody
		(
			new
			{
				token = WebFrameworkUtility.CreateJwt(temp, TimeSpan.FromMinutes(60))
			}
		);
	}
}
